src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -28,14 +34,49 @@
      * 26.04.2019 Сергей 7 100
      * 26.04.2019 София 9 100
      * 26.04.2019 Сергей 11 50
-     *
+     * </p>
      * Пример вывода:
      * Отчёт за период 01.04.2019  - 30.04.2019
      * Сергей - 1550
      * Андрей - 600
      * София - 900
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder allSalary = new StringBuilder();
+        allSalary.append("Отчёт за период " + dateFrom + " - " + dateTo + "\n");
+        DateFormat format = new SimpleDateFormat("dd.MM.yyyy", Locale.ENGLISH);
+        int salary1 = 0;
+        int salary2 = 0;
+        int salary3 = 0;
+
+        try {
+            Date minDate = format.parse(dateFrom);
+            Date maxDate = format.parse(dateTo);
+            if (minDate.compareTo(maxDate) > 0) {
+                return null;
+            }
+
+            for (String s : data) {
+                String[] workInfo = s.split(" ");
+                Date workDate = format.parse(workInfo[0]);
+                if (workDate.compareTo(minDate) >= 0 && workDate.compareTo(maxDate) <= 0) {
+                    salary1 += workInfo[1].equals(names[0])
+                            ? (Integer.parseInt(workInfo[workInfo.length - 2]))
+                            * (Integer.parseInt(workInfo[workInfo.length - 1])) : 0;
```

Comment:
`salary1 += workInfo[1].equals(names[0])
                            ? (Integer.parseInt(workInfo[workInfo.length - 2]))
                            * (Integer.parseInt(workInfo[workInfo.length - 1])) : 0;`
this is hard to read. Please create a tmp variables and refactor this. Also you can declare the 0, 1, 2 ... as a constants

Reply:
fixed